### PR TITLE
[#22660] Upgrade org.apache.samza to 1.7

### PR DESCRIPTION
**Pull request by kw2542**
1. Upgrade org.apache.samza to 1.7
2. Enhance SamzaServerConfiguration to be configurable of base dir for states in order to isolate dir for different jobs to avoid state conflict.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kw2542** [comment]
Run Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

File: runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobServerDriver.java [hunk 1]
```diff
@@ -17,23 +17,52 @@
...
 
   private static final Logger LOG = LoggerFactory.getLogger(SamzaJobServerDriver.class);
 
   /** Samza runner-specific Configuration for the jobServer. */
-  public static class SamzaServerConfiguration extends ServerConfiguration {}
+  public static class SamzaServerConfiguration extends ServerConfiguration {
+    @Option(
+        name = "--job-non-logged-store-base-dir",
+        usage =
+            "Base dir for non logger store, job name will be automatically "
```
Comment (dxichen) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobServerDriver.java:45]: s/logger/logged

File: runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobServerDriver.java [hunk 2]
```diff
@@ -17,23 +17,52 @@
...
+
+    @Nullable
+    String getJobNonLoggedStoreBaseDir() {
+      return jobNonLoggedStoreBaseDir;
+    }
+
+    @Option(
+        name = "--job-logged-store-base-dir",
+        usage =
+            "Base dir for logger store, job name will be automatically "
```
Comment (dxichen) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobServerDriver.java:57]: s/logger/logged

**Review by dxichen — APPROVED** [review]
Thanks for the quick fix, lgtm after the typos
Could you also update the PR title and desc as well?

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

**Comment by kw2542** [comment]
Run Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Python Samza ValidatesRunner

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

**Comment by kw2542** [comment]
Run Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Python Samza ValidatesRunner

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Python Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Java PreCommit

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Java PreCommit

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Java PreCommit

**Comment by jrmccluskey** [comment]
The Go ValidatesRunner suite seems to be failing with timeouts for each test.

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Go Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Go PreCommit

**Comment by kw2542** [comment]
Run Java Samza PortableValidatesRunner

**Comment by kw2542** [comment]
Run Samza ValidatesRunner

**Comment by kw2542** [comment]
Run Python Samza ValidatesRunner

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run Java PreCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run XVR_Samza PostCommit

**Comment by kw2542** [comment]
Run Go PreCommit

**Comment by kw2542** [comment]
Run Python PreCommit

**Pull request closed without merging**